core/region.go

```diff
@@ -474,3 +476,54 @@ func (r *region) findEnabledASGByName(name string) *autoScalingGroup {
 	}
 	return nil
 }
+
+func (r *region) sqsSendMessageSpotInstanceLaunch(asgName *string, instanceID *string, instanceState *string) error {
+	inputJSON := "{\"version\":\"0\",\"id\":\"890abcde-f123-4567-890a-bcdef1234567\"," +
```

Comment:
I don't like this defined inline, can we convert it to a Go template?

I'm also unclear on why we even need to do this.

Reply:
i replaced all ondemand replacement with a call that send a message to the sqsqueue.
The message body is equals to the cloudwatch event of the relative spot instance start.
I just need:
* time, so that sqs content deduplication can work (two message for the same instance are treated as different one)
* instanceID
* instanceState
* region

other field are not used.